### PR TITLE
Streamswitcher

### PR DESCRIPTION
**Pull request by CodeCasterNL**
Quick & dirty implementation of #67. Not sure whether the `async void` is what you'd want, and setting the time of the previous stream to the current doesn't seem to work.

This is extremely untested, more like a POC whether this would be the way to go. Not sure if this properly disposes what needs disposing, and whether non-live / non-multistream episodes and sessions keep working.

### CODE REVIEW

File: RaceControl/RaceControl/ChannelCollection.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace RaceControl
+{
+    public class ChannelCollection : BindableBase, IChannelCollection
+    {
+        public ChannelCollection(ObservableCollection<IPlayableChannel> allChannels, IPlayableChannel currentChannel)
+        {
+            CurrentChannel = currentChannel;
+
+            Onboards = new ObservableCollection<IPlayableChannel>(allChannels.Where(c => c.ChannelType == ChannelType.Onboard));
```
Comment (CodeCasterNL) [inline RaceControl/RaceControl/ChannelCollection.cs:15]: Perhaps also sort drivers by driver number or last name? But we only get a single string here, `c.DisplayName`.

**Comment by bootsie123** [comment]
I would love to see this implemented. Any updates?

**Comment by robvdpol** [comment]
> I would love to see this implemented. Any updates?

Other things first, I'm looking into moving away from VLC for the internal player since it's so unreliable for live streams.

**Comment by bootsie123** [comment]
ah, no worries! Keep up the great work!

**Comment by CodeCasterNL** [comment]
I've merged master to support the new FlyLeaf player and the switch to context menus, but there are now two issues:

* Audio streams don't reload after changing stream
* Setting the time from the previous stream to the current still doesn't work; stream starts at 0

**Comment by CodeCasterNL** [comment]
@robvdpol I've fixed the double event subscriptions, and now the new channel continues at the current time. Also set "und" to default channel, quick-fixes #108 as well.

**Comment by CodeCasterNL** [comment]
@robvdpol merged master again, but now it's FUBAR: FlyLeaf throws a PlatformNotSupportedException on `FlyLeafMediaPlayer.Stop()` after 30 seconds of hanging, as well as on `FlyLeafMediaPlayer.Open(string)` if you catch the former.

You also can't dispose the player and re-initialize it using a factory (`Func<Player>` as constructor parameter), as that will throw NullReferenceExceptions while the player window is playing something.

Back to the drawing board it is.

**Comment by robvdpol** [comment]
> @robvdpol merged master again, but now it's FUBAR: FlyLeaf throws a PlatformNotSupportedException on `FlyLeafMediaPlayer.Stop()` after 30 seconds of hanging, as well as on `FlyLeafMediaPlayer.Open(string)` if you catch the former.
> 
> You also can't dispose the player and re-initialize it using a factory (`Func<Player>` as constructor parameter), as that will throw NullReferenceExceptions while the player window is playing something.
> 
> Back to the drawing board it is.

Don't worry about it, I'll have a look soon. New mediaplayer is reasonably stable now so I can focus on other stuff again.

**Pull request closed without merging**